libs/vkd3d/command.c

```diff
@@ -3263,7 +3263,8 @@ static bool d3d12_command_list_gather_pending_queries(struct d3d12_command_list
     for (i = 0; i < dispatch_count; i++)
     {
         struct dispatch_entry *d = &dispatches[i];
-        memset(query_map, 0, sizeof(*query_map) * query_map_size);
+        if (i != 0)
+            memset(query_map, 0, sizeof(*query_map) * query_map_size);
```

Comment:
why is this better than the current code? *Something* needs to zero-initialize the memory in question anyway so this doesn't make a huge amount of sense to me.

And if we want to overcomplicate this for no reason, might as well clear the query map at the end rather than the beginning of the loop.

Reply:
calloc doesn't necessarily memset. If it calls mmap to allocate, it can assume the page is already pre zero'ed and do nothing. So it's always better to use calloc instead of malloc + memset.

Reply:
Yeah that would be ideal, in which case this would have to be reverted anyway.

Reply:
From Discord, it seems like this wasn't flagged as a significant perf issue, just an observation (I think I misunderstood the discussion), should we table this for later? It seems like the correct solution is a rethink of the algorithm at a higher level.